ddl/db_test.go

```diff
@@ -2656,44 +2656,66 @@ func (s *testDBSuite4) TestAddColumn2(c *C) {
 	s.tk.MustQuery("select a,b,_tidb_rowid from t2").Check(testkit.Rows("1 3 2"))
 }
 
+// --------------------------------------------------------
+//func (s *testDBSuite5) TestAddIndexForGeneratedColumn(c *C) {
+//	s.tk = testkit.NewTestKit(c, s.store)
+//	s.tk.MustExec("use test_db")
+//	s.tk.MustExec("create table t(y year NOT NULL DEFAULT '2155')")
+//	defer s.mustExec(c, "drop table t;")
+//	for i := 0; i < 50; i++ {
+//		s.mustExec(c, "insert into t values (?)", i)
+//	}
+//	s.tk.MustExec("insert into t values()")
+//	s.tk.MustExec("ALTER TABLE t ADD COLUMN y1 year as (y + 2)")
+//	_, err := s.tk.Exec("ALTER TABLE t ADD INDEX idx_y(y1)")
+//	c.Assert(err.Error(), Equals, "[ddl:15]cannot decode index value, because cannot convert datum from unsigned bigint to type year.")
+//
+//	t := s.testGetTable(c, "t")
+//	for _, idx := range t.Indices() {
+//		c.Assert(strings.EqualFold(idx.Meta().Name.L, "idx_c2"), IsFalse)
+//	}
+//	s.mustExec(c, "delete from t where y = 2155")
+//	s.mustExec(c, "alter table t add index idx_y(y1)")
+//	s.mustExec(c, "alter table t drop index idx_y")
+//
+//	// Fix issue 9311.
+//	s.tk.MustExec("create table gcai_table (id int primary key);")
```

Comment:
No need these tests anymore?

Reply:
The behaviors it tests are what we actually forbid, so I refactor this function. PTAL at the line 2700.

Reply:
The behaviors it tests are what we actually forbid, so I refactor this function. PTAL at the line 2700.